docarray/array/array/io.py

```diff
@@ -302,6 +304,56 @@ def to_json(self) -> str:
         """
         return json.dumps([doc.json() for doc in self])
 
+    @classmethod
+    def _check_for_valid_document_type(cls) -> None:
+        if cls.document_type == AnyDocument:
```

Comment:
is there are plan to make this check more sophisticated as part of this PR? otherwise this doesn't need to be a method imo